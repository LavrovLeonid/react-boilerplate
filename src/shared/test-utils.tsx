import { FC } from 'react';
import { BrowserRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies -- Не используется в продакшене
import { render } from '@testing-library/react';
// eslint-disable-next-line import/no-extraneous-dependencies -- Не используется в продакшене
import userEvent from '@testing-library/user-event';

const AllTheProviders: FC = ({ children }) => (
  <BrowserRouter>{children}</BrowserRouter>
);

const customRender = (
  ui: Parameters<typeof render>[0],
  options?: Parameters<typeof render>[1],
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderWithRouter = (
  ui: Parameters<typeof render>[0],
  { route = '/' } = {},
) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: AllTheProviders }),
  };
};

// eslint-disable-next-line import/no-extraneous-dependencies -- Не используется в продакшене
export * from '@testing-library/react';
export { customRender as render, renderWithRouter };
