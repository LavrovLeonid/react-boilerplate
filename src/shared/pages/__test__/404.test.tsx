import { render } from '@shared/test-utils';
import NotFoundPage from '../404';

test('404 page', () => {
  const { container } = render(<NotFoundPage />);

  expect(container.firstChild).toMatchSnapshot();
});
