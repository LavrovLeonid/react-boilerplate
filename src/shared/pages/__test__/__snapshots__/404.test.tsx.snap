// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 page 1`] = `
<div>
  not found
</div>
`;
